### PR TITLE
Receive and update waveforms through websocket

### PR DESCRIPTION
**Pull request by tannermpeterson**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,5 +1,12 @@
 Changelog for Mantarray Frontend Components
 ===========================================
+
+0.3.0 (unreleased)
+------------------
+
+- Added heatmap
+- Added websocket to receive waveform data asynchronously and in real time
+
```
Comment (tannermpeterson) [inline CHANGELOG.rst:9]: Not sure what else needs to be put in this, I think @jperson mentioned there were some new y-axis zoom features that have been implemented

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@curi-bio/mantarray-frontend-components",
-  "version": "0.6.0",
+  "version": "0.3.0",
```
Comment (tannermpeterson) [inline package.json:3]: Last release was `0.2.0`, so we can still mark this version as `0.3.0`

File: CHANGELOG.rst (hunk 1 above)
Comment (jperson) [reply CHANGELOG.rst:9]: @tannermpeterson yeah, it can say `Added y-axis-zoom for waveform display`

File: package.json (hunk 2 above)
Comment (luciipak) [reply package.json:3]: Got it, that's my bad, I thought every big, unbroken merge required an increase in the minor version

File: README.rst [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
 
+Install steps:
+ - ``npm install``
+ - ``npm run build``
+
+Good to confirm that Unit Tests are all passing after install:
+ - ``npm run test:unit``
+
+Load and view a page after running command ``npm run dev``
+
```
Comment (luciipak) [inline README.rst:11]: this is great, thanks

File: CHANGELOG.rst (hunk 1 above)
Comment (luciipak) [reply CHANGELOG.rst:10]: I also need to add to this..

**Review by luciipak — APPROVED** [review]

File: tests/unit/store/waveform.spec.js [hunk 4]
```diff
@@ -107,6 +110,56 @@ describe("store/waveform", () => {
...
+    let ws_server;
+    let socket_client;
+
+    beforeAll((done) => {
+      http_server = http.createServer().listen(4567); // TODO use constant here
+      ws_server = io_server(http_server);
+      done();
+    });
+
+    beforeEach(function (done) {
```
Comment (luciipak) [inline tests/unit/store/waveform.spec.js:124]: I would stick to the JS ES6+ arrow functions that you used above in the beforeAll (x => {}) (const x = () => {}),

What are the done callbacks used for?

Comment (tannermpeterson) [reply tests/unit/store/waveform.spec.js:133]: These callbacks are probably unnecessary except for in `beforeEach` where they make sure a connection is made before starting the test

**Review by jperson — APPROVED** [review]

**Pull request merged**